tools/webpack/webpack.renderer.config.js

```diff
@@ -0,0 +1,80 @@
+const rules = require('./common/webpack.rules');
+const plugins = require('./common/webpack.plugins');
+const MonacoWebpackPlugin = require('monaco-editor-webpack-plugin');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+
+plugins.push(
+  new MonacoWebpackPlugin({
+    languages: ['typescript', 'javascript', 'html', 'css'],
+  }),
+);
+plugins.push(
+  new MiniCssExtractPlugin({
+    filename: './css/[name].css',
+    // https://github.com/webpack-contrib/mini-css-extract-plugin#experimentalUseImportModule
+    experimentalUseImportModule: false,
+  }),
+);
+
+// Handling style files
+rules.push({
+  test: /\.css$/,
+  use: [{ loader: MiniCssExtractPlugin.loader }, { loader: 'css-loader' }],
+});
+
+rules.push({
+  test: /\.less$/i,
+  use: [
+    {
+      loader: MiniCssExtractPlugin.loader,
+    },
+    {
+      loader: 'css-loader',
+    },
+    {
+      // Used to load the url loaders present in blueprintjs (/resources/icons)
+      loader: 'resolve-url-loader',
+    },
+    {
+      loader: 'less-loader',
+      options: {
+        lessOptions: {
+          // Used to evaluate css function (https://github.com/palantir/blueprint/issues/5011).
+          math: 'always',
+        },
+      },
+    },
+  ],
+});
+
+// Handling assets
+rules.push({
+  test: /\.(jpe?g|svg|png|gif|ico)(\?v=\d+\.\d+\.\d+)?$/i,
+  type: 'asset/resource',
+  generator: {
+    filename: 'assets/[name][ext]',
+  },
+});
+
+// Fonts
+rules.push({
+  test: /\.(eot|ttf|woff2?)(\?v=\d+\.\d+\.\d+)?$/i,
+  type: 'asset/resource',
+  generator: {
+    filename: 'fonts/[name][ext]',
+  },
+});
```

Comment:
Let's change these. The current code, pushing into the array exported by the other modules, it actually mutating the array value. That's generally not a good idea as it can introduce subtle bugs (any other import after this one will see the mutated value).

Let's use spread syntax to accomplish this without mutating `rules` and `plugins`. It would look something like this:

```
module.exports = {
  module: {
    rules: [
        ..rules,
        {
            test: /\.(eot|ttf|woff2?)(\?v=\d+\.\d+\.\d+)?$/i,
            type: 'asset/resource',
            generator: {
                filename: 'fonts/[name][ext]',
            },
            // etc
        }
    ]
  },
  plugins: [
      ...plugins,
      new MonacoWebpackPlugin({
          languages: ['typescript', 'javascript', 'html', 'css'],
      }),
      // etc
  ]
```

Reply:
Yup, Sounds right! I was only doing that in only renderer process as it's the last piece of code that executes :)